Sources/KSCrashRecording/include/KSCrashConfiguration.h

```diff
@@ -0,0 +1,149 @@
+    KSCDeleteNever,
+    KSCDeleteOnSucess,
+    KSCDeleteAlways
+};
+
+@interface KSCrashConfiguration : NSObject <NSCopying>
+
+/** The crash types that will be handled.
+ * Some crash types may not be enabled depending on circumstances (e.g., running in a debugger).
+ * 
+ * **Default**: `KSCrashMonitorTypeProductionSafeMinimal`
+ */
+@property (nonatomic, assign) KSCrashMonitorType monitors;
+
+/** User-supplied data in JSON format. NULL to delete.
+ *
+ * This JSON string contains user-specific data that will be included in
+ * the crash report. If NULL is passed, any existing user data will be deleted.
+ */
+@property (nonatomic, copy, nullable) NSDictionary <NSString*, id>* userInfoJSON;
+
+/** The maximum time to allow the main thread to run without returning.
+ *
+ * If the main thread is occupied by a task for longer than this interval, the
+ * watchdog will consider the queue deadlocked and shut down the app, writing a
+ * crash report. Set to 0 to disable this feature.
+ *
+ * **Warning**: Ensure that no tasks on the main thread take longer to complete than
+ * this value, including application startup. You may need to initialize your
+ * application on a different thread or set this to a higher value until initialization
+ * is complete.
+ */
+@property (nonatomic, assign) double deadlockWatchdogInterval;
+
+/** If true, attempt to fetch dispatch queue names for each running thread.
+ *
+ * This option enables the retrieval of dispatch queue names for each thread at the
+ * time of a crash. This can provide useful context, but there is a risk of crashing
+ * during the `ksthread_getQueueName()` call.
+ *
+ * **Default**: false
+ */
+@property (nonatomic, assign) BOOL enableQueueNameSearch;
+
+/** If true, introspect memory contents during a crash.
+ *
+ * Enables the inspection of memory contents during a crash. Any Objective-C objects
+ * or C strings near the stack pointer or referenced by CPU registers or exceptions
+ * will be included in the crash report, along with their contents.
+ *
+ * **Default**: false
+ */
+@property (nonatomic, assign) BOOL enableMemoryIntrospection;
+
+/** List of Objective-C classes that should never be introspected.
+ *
+ * A list of class names that should not be inspected during a crash. Only the class
+ * names will be recorded in the crash report when instances of these classes are
+ * encountered. This is useful for information security.
+ *
+ * **Default**: NULL
+ */
+@property (nonatomic, strong, nullable) NSArray<NSString *> *doNotIntrospectClasses;
+
+/** Callback to invoke upon a crash.
+ *
+ * This function is called during the crash reporting process, providing an opportunity
+ * to add additional information to the crash report. Only async-safe functions should
+ * be called from this function. Avoid calling Objective-C methods.
+ *
+ * **Default**: NULL
+ */
+@property (nonatomic, copy, nullable) void (^crashNotifyCallback)(const struct KSCrashReportWriter* writer);
```

Comment:
This is the "ugly" API for Swift. `KSCrashReportWriter` is a struct that deeply integrated in KSCrash and used in callback, that makes it hard to proxy. 
Visible to Swift:
```swift
open var crashNotifyCallback: ((UnsafePointer<KSCrashReportWriter>) -> Void)?
```

Reply:
Wrapping this in an ObjC class would be a really good thing to have. Swift API matters.

Reply:
From docs
```
 * WARNING: Only call async-safe functions from this function! DO NOT call
 * Objective-C methods!!!
 ```

Reply:
Right. Well, it's fine then.
I'm thinking that a wrapper with all methods being `__attribute__((objc_direct))` might do the trick, but it's still ObjC.